src/index.ts

```diff
@@ -373,6 +382,27 @@ export const useForm = <Values extends Record<string, unknown>, ErrorMessage = s
       return Promise.resolve(internalValidateField(name));
     };
 
+    const listenFields: ListenFieldsFunction<Values, ErrorMessage> = (names, onUpdate) => {
+      const callback = () => {
```

Comment:
No need for variable declaration + you are in the `api` scope, no need for `api.xxx`

```ts
    const listenFields: Contract["listenFields"] = (names, listener) => {
      const callback = () => {
        listener(
          names.reduce((acc, name) => {
            acc[name] = transformState(name, states.current[name], { sanitize: false });
            return acc;
          }, {} as { [N1 in typeof names[number]]: FieldState<Values[N1], ErrorMessage> })
        );
      };

      names.forEach((name) => callbacks.current[name].add(callback));

      return () => {
        names.forEach((name) => callbacks.current[name].delete(callback));
      };
    };
```

Reply:
I like create variables, it helps me for debugging, but this way is good for me. I made the change